addons/website/controllers/main.py

```diff
@@ -619,6 +619,14 @@ def pagenew(self, path="", noredirect=False, add_menu=False, template=False, **k
         template = template and dict(template=template) or {}
         page = request.env['website'].new_page(path, add_menu=add_menu, **template)
         url = page['url']
+        # In case the page is created through the 404 "Create Page" button, the
+        # URL may use special characters which are slugified on page creation.
+        # If that URL is also a menu, we update it accordingly.
+        # NB: we don't want to slugify on menu creation as it could redirect
+        # towards files (with spaces, apostrophes, etc.).
+        menu = request.env['website.menu'].search([('url', '=', '/' + path)])
+        if menu:
+            menu.url = url
```

Comment:
@robinlej It works but I wonder if you'd rather not link the menu to the created page through the page_id field?
![image](https://github.com/odoo/odoo/assets/10338094/0e120166-fc30-40a9-b335-48da68bc28ea)

Then the URL sync would be automatic? Though I am not sure, to test. URL should really be a computed field... :grimacing:

Reply:
URL is not exactly a computed field, but you're right and the syncing works out of the box.
Follow-up PR here: https://github.com/odoo/odoo/pull/165616

Reply:
> URL is not exactly a computed field

I know, I was saying that it should really be in a future refactoring :)

> Follow-up PR here: https://github.com/odoo/odoo/pull/165616

Thanks :+1: